app/uk/gov/hmrc/transitmovementsvalidator/services/XmlValidation.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.transitmovementsvalidator.services
+
+import uk.gov.hmrc.transitmovementsvalidator.models.MessageTypeXml
+
+import javax.xml.XMLConstants
+import javax.xml.parsers.SAXParserFactory
+import javax.xml.validation.SchemaFactory
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+trait XmlValidation {
+
+  def parsersByType(messageType: MessageTypeXml)(implicit ec: ExecutionContext): Future[SAXParserFactory] =
```

Comment:
This should be a `val` that contains the Map. Here, you're recreating the map every time, so you're building the parser every time, something we shouldn't be doing.

With the `Map[MessageType, Future[SAXParserFactory]]`, you can then just `parsersByType.get(MessageType)` which returns an `Option`.